src/watchmaker/workers/salt.py

```diff
@@ -568,7 +581,8 @@ def __init__(self, *args, **kwargs):
         self.salt_version = kwargs.pop('salt_version', None) or ''
 
         # Enforce lowercase and replace spaces with ^ in Linux
-        self.admin_groups = self.admin_groups.lower().replace(' ', '^')
+        if hasattr(self.admin_groups, 'lower'):
```

Comment:
ok, `None` would be a problem. coerce the default to the expected type in the `__init__` method... e.g.

```
        self.admin_groups = watchmaker.utils.config_none_deprecate(
            self.admin_groups, self.log) or ''
```

Reply:
I like Pythonic `None` better than an empty string. This should be fixable with just checking to see if `admin_groups`. If so, it will be a string.